apps/server/src/modules/idp-console/idp-console.module.ts

```diff
@@ -0,0 +1,81 @@
+import { Module } from '@nestjs/common';
+import { ConsoleModule } from 'nestjs-console';
+import { CqrsModule } from '@nestjs/cqrs';
+import { ConsoleWriterModule } from '@infra/console';
+import { ConfigModule } from '@nestjs/config';
+import { HttpModule, HttpService } from '@nestjs/axios';
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+import { Logger, LoggerModule } from '@src/core/logger';
+import { MikroOrmModule } from '@mikro-orm/nestjs';
+import { DB_PASSWORD, DB_URL, DB_USERNAME } from '@src/config';
+import { SchulconnexRestClient } from '@infra/schulconnex-client';
+import { AccountModule } from '@modules/account';
+import { OauthAdapterService } from '@modules/oauth';
+import { defaultMikroOrmOptions } from '@modules/server';
+import { ALL_ENTITIES } from '@shared/domain/entity';
+import { UserService } from '@modules/user';
+import { UserRepo } from '@shared/repo';
+import { UserDORepo } from '@shared/repo/user/user-do.repo';
+import { RoleModule } from '@modules/role';
+import { IdentityManagementModule } from '@infra/identity-management';
+import { RegistrationPinModule } from '@modules/registration-pin';
+import {
+	SynchronizationEntity,
+	SynchronizationRepo,
+	SynchronizationService,
+	SynchronizationUc,
+} from '@modules/synchronization';
+import { IdpSyncConsole } from './idp-sync-console';
+
+@Module({
+	imports: [
+		ConfigModule.forRoot({ isGlobal: true }),
+		ConsoleModule,
+		ConsoleWriterModule,
+		LoggerModule,
+		CqrsModule,
+		MikroOrmModule.forRoot({
+			...defaultMikroOrmOptions,
+			type: 'mongo',
+			clientUrl: DB_URL,
+			password: DB_PASSWORD,
+			user: DB_USERNAME,
+			allowGlobalContext: true,
+			entities: [...ALL_ENTITIES, SynchronizationEntity],
+			debug: true,
+		}),
+		IdentityManagementModule,
+		HttpModule,
+		AccountModule,
+		RegistrationPinModule,
+		RoleModule,
+	],
+	providers: [
+		SynchronizationRepo,
+		SynchronizationService,
+		SynchronizationUc,
+		OauthAdapterService,
+		IdpSyncConsole,
+		UserDORepo,
+		UserRepo,
+		UserService,
```

Comment:
dont provide the UserService again, import the userModule instead.

Reply:
thats strange, in the adminapi deployment we use the UserAdminApiModule, which imports the UserModule, and it isnt an issue there.

It might be worth investigating the difference. Maybe it has to do with the way the ConfigModule is set up?

Anyway, its much better to add any necessary dependencies to the new deployment, than to tangle modules in more obfuscated ways.

That some modules, including the UserModule, might have too many not strictly necessary dependencies is known, and fixing that in the future will automatically reduce the footprint of your application.

Also from a quick glance, I didnt see why the UserModule needs the FilesStorageProducer. Have you seen where exactly thats coming from?

Reply:
Fixed in https://github.com/hpi-schul-cloud/schulcloud-server/pull/4859/commits/af359ebeab64a04dfeec65f8b6d1ce24170ae995 .

Regarding this `FilesStorageProducer` (and `AmqpConnection` dependency) - I'm still not entirely sure why is it needed, but I've managed to fix it simply by importing the `RabbitMQWrapperModule` module.